### ISSUE TITLE
Keyword Input Arguments for Click Events

### ISSUE THREAD

**Issue by EntilZha** [gradio-app/gradio#5049]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
I am wrapping gradio around a model function with many keyword arguments or some keyword only arguments. Unfortunately, it seems like gradio click events like `gradio.Button.click` https://www.gradio.app/docs/button, only support positional arguments. My current workaround is to write a wrapper, but it would be great to be able to directly use the function.

**Describe the solution you'd like**  
I would like for `gradio.Button.click` to accept by some means keyword arguments to its associated `fn`. The easiest solution is to add an extra argument `input_kwargs`, similar how to Streamlit solves the same issue https://docs.streamlit.io/library/api-reference/widgets/st.button. For example, it would be great to have it work like this:

```python
import gradio as gr


def greet(name: str, *, last_name: str):
    return name + " " + last_name

with gr.Blocks() as demo:
    txt_0 = gr.Text()
    txt_1 = gr.Text()
    txt_2 = gr.Text()
    btn = gr.Button("Click me")
    # This is the change, allow inputs be a positional argument list and inputs_kwargs be a dictionary of args
    btn.click(greet, inputs=[txt_0], inputs_kwargs={'last_name': txt_1}, outputs=[txt_2])


if __name__ == '__main__':
    demo.launch(server_name="0.0.0.0", server_port=7860)
```

**Comment by abidlabs** [comment]
Yeah this is a good suggestion. I think it would require quite a bit of work. We'll look into it after 4.0. 

A related issue is: #4857

**Comment by EntilZha** [comment]
That would be great! By chance any ideas for workaround for the time being?

**Comment by JackismyShephard** [comment]
I think this would still be highly appreciated, especially if it could work with type checkers, which the current positional approach does not.

**Comment by AryaLolusare2712** [comment]
Hi! I'd like to work on this feature if it's still available. I plan to prototype support for inputs_kwargs while keeping full backward compatibility with the existing inputs API. If this approach aligns with the maintainers' expectations, I'd be happy to open a PR.
